addons/crm/tests/test_crm_lead.py

```diff
@@ -294,3 +294,72 @@ def test_mailgateway(self):
         new_lead._handle_partner_assignment(create_missing=True)
         self.assertEqual(new_lead.partner_id.email, 'unknown.sender@test.example.com')
         self.assertEqual(new_lead.partner_id.team_id, self.sales_team_1)
+
+    def check_phone_domain(self, leads, term, mapping, indexes = None):
+
+        domain = leads._search_phone_mobile_search('ilike', term)
+        if not len(domain) == 1:
+            self.assertTrue(False) # Fail
+            return
+
+        if indexes is None:
+            for (name, operator, ids) in domain:
+                if operator == 'in':
+                    self.assertFalse(any(elem in ids for elem in mapping))
+                else:
+                    self.assertEqual(domain, [(0, '=', 1)])
+            return
+
+        for (name, operator, ids) in domain:
+            self.assertEqual(name, 'id')
+            self.assertEqual(operator, 'in')
+            self.assertEqual(
+                list(filter(lambda elem: elem in mapping, ids)).sort(),
+                [mapping[k] for k in indexes].sort()
+            )
+
+    def test_phone_filter(self):
+
+        phones = [
+            '754-3010',
+            '(541) 754-3010',
+            '+1-541-754-3010',
+            '1-541-754-3010',
+            '001-541-754-3010',
+            '191 541 754 3010',
+            '636-48018',
+            '(089) / 636-48018',
+            '+49-89-636-48018',
+            '19-49-89-636-48018'
+        ]
+
+        leads_data = self._create_leads_data_batch(count=len(phones))
+        for (phone, lead_data) in zip(phones, leads_data):
+            lead_data.update({
+                "phone": phone
+            })
+
+        leads = self.env['crm.lead'].create(leads_data)
+        mapping = [lead.id for lead in leads]
+
+        self.check_phone_domain(leads, '754', mapping, list(range(0, 6)))
```

Comment:
The way you built the test makes it a bit hard to understand what's the input and what's the expected output.
Which is usually how tests are written and easy to understand / modify.

So apart from the pre-step of constructing your leads dataset, I would prefer something like:
```py

searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '11')])
# should not return anything, search parameter should be at least 3 chars
self.assertFalse(bool(searched_leads))

searched_leads = self.env['crm.lead'].search([('phone_mobile_search', 'like', '+32499')])
# should return both '32499112233' and '0032499113344' since '+32' and '00' prefix are equal
self.assertEqual(['+32499112233', '0032499113344'], search_leads.mapped('phone'))

# then do the same but search with 0032499
```

The goal is really only to test that your regexes and checks are producing the correct result.
Including special cases like "+32" should be the same as "00".
Then you can try maybe with the prefix from France to make sure (and then USA prefixes maybe?).
And then include some of your own use cases from your current code.